can_tools/scrapers/official/MO/mo_state.py

```diff
@@ -0,0 +1,233 @@
+from abc import ABC
+
+import pandas as pd
+
+import pyppeteer
+import us
+from typing import List
+
+from can_tools.scrapers.base import CMU
+from can_tools.scrapers.official.base import StateDashboard
+from can_tools.scrapers.puppet import TableauNeedsClick
+
+
+class MissouriBase(
+    StateDashboard,
+    TableauNeedsClick,
+    ABC,
+):
+    source = "https://results.mo.gov/t/COVID19/views/COVID-19DataforDownload/{tab}?:isGuestRedirectFromVizportal=y&:embed=y"
+    has_location = False
+    location_type = "county"
+    state_fips = int(us.states.lookup("Missouri").fips)
+
+    out_cols = [
+        "dt",
+        "vintage",
+        "location_name",
+        "category",
+        "measurement",
+        "unit",
+        "age",
+        "race",
+        "sex",
+        "value",
+    ]
+
+    # def clean_desoto(self, df: pd.DataFrame):
+    #    df.loc[df["location_name"] == "Desoto", "location_name"] = "DeSoto"
+
+
+async def _special_tableau_handler(page, tmpdirname):
+    # Get a reference to the page that was just opened
+    pages = await page.browser.pages()
+    popup = pages[-1]
+
+    import requests
+    import os
+
+    await popup.waitForSelector(".csvLink_summary")
+    url = await popup.querySelectorEval(".csvLink_summary", "node => node.href")
+    data = requests.get(url)
+    open(
+        os.path.join(tmpdirname, "Metrics_by_Test_Date_by_County_data.csv"), "wb"
+    ).write(data.content)
+
+
+class MissouriState(MissouriBase):
+    """
+    Fetch details about overall Missouri state testing by county
+    """
+
+    headless = True
+    url = MissouriBase.source.format(tab="MetricsbyTestDatebyCounty")
+
+    # TODO0: ZW: The Data tab below is better... I think. Remove this once confirmed which data to use
+    #    # Download Crosstab from Tableau
+    #    tableau_click_selector_list: List[str] = [
+    #        "@data-tb-test-id = 'DownloadCrosstab-Button'",
+    #        "@data-tb-test-id = 'crosstab-options-dialog-radio-csv-Label'",
+    #        "@data-tb-test-id = 'export-crosstab-export-Button'"
+    #    ]
+
+    # Download Data from Tableau
+    tableau_click_selector_list: List[str] = [
+        "@data-tb-test-id = 'DownloadData-Button'",
+        _special_tableau_handler,
+    ]
+
+    async def _pre_click(self, page: pyppeteer.page.Page):
+        # TODO0: ZW: This page is slow to load. Adding wait.
+        #  Should probably find a better selector to use to determine that the page has loaded
```

Comment:
It looks like you wait for `#main-content` does that remove the need for sleeping for 5?

Reply:
It is going to have to be one or the other. That's the TODO, get back in there and figure out if I can find a css selector that will indicate the page, and more importantly, the data is loaded. There looks like there is a Tableau spinning arrow that pops up when the data is loading and goes away when it is ready. I am going to see if I can trigger off that.